packages/graphql/lib/fragments.js

```diff
@@ -1,49 +1,66 @@
+/* global fetch */
 'use strict';
 
+require('isomorphic-fetch');
 var fs = require('fs');
-
-var fetch = require('isomorphic-fetch');
+var pify = require('pify');
+var graphql = require('graphql').graphql;
+var makeExecutableSchema = require('graphql-tools').makeExecutableSchema;
 
 var hopsConfig = require('hops-config');
 
 var fragmentsFile = require('./util').getFragmentsFile();
 
-module.exports = function fetchFragments() {
-  return fetch(hopsConfig.graphqlUri, {
+function writeFragmentTypesFile(result) {
+  result.data.__schema.types = result.data.__schema.types.filter(function(t) {
+    return t.possibleTypes !== null;
+  });
+  return pify(fs.writeFile)(fragmentsFile, JSON.stringify(result.data));
+}
+
+function executeRemoteQuery(graphqlUri, query) {
```

Comment:
Some users might need to add additional headers/params to authenticate with their graphQL endpoint.

Do you think it'd be helpful to allow that in any way?